src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+          symbol.io === IoInput &&
+          symbol.scope === call.fullyQualifiedName).result
+
+      symbolStoreEntries = symbolResults map { symbolResult =>
+        val wdlType = WdlType.fromWdlString(symbolResult.wdlType)
+        val symbolStoreKey = new SymbolStoreKey(
+          symbolResult.scope,
+          symbolResult.name,
+          symbolResult.iteration,
+          symbolResult.io == IoInput)
+        new SymbolStoreEntry(
+          symbolStoreKey,
+          wdlType,
+          Option(symbolResult.wdlValue).map(wdlType.coerceRawValue(_).get))
+
+      }
+    } yield symbolStoreEntries
+
+    runTransaction(action)
+  }
+
+  /** Get all outputs for the scope of this call. */
+  override def getOutputs(workflowId: WorkflowId, call: Call): Future[Traversable[SymbolStoreEntry]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      symbolResults <- dataAccess.symbols.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId &&
+          symbol.io === IoOutput &&
+          symbol.scope === call.fullyQualifiedName).result
+
+      symbolStoreEntries = symbolResults map { symbolResult =>
+        val wdlType = WdlType.fromWdlString(symbolResult.wdlType)
+        val symbolStoreKey = new SymbolStoreKey(
+          symbolResult.scope,
+          symbolResult.name,
+          symbolResult.iteration,
+          symbolResult.io == IoInput)
+        new SymbolStoreEntry(
+          symbolStoreKey,
+          wdlType,
+          Option(symbolResult.wdlValue).map(wdlType.coerceRawValue(_).get))
+
+      }
+
+    } yield symbolStoreEntries
+
+    runTransaction(action)
+  }
+
+  /** Returns all outputs for this workflowId */
+  override def getOutputs(workflowId: WorkflowId): Future[Traversable[SymbolStoreEntry]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      symbolResults <- dataAccess.symbols.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId &&
+          symbol.io === IoOutput).result
+
+      symbolStoreEntries = symbolResults map { symbolResult =>
+        val wdlType = WdlType.fromWdlString(symbolResult.wdlType)
+        val symbolStoreKey = new SymbolStoreKey(
+          symbolResult.scope,
+          symbolResult.name,
+          symbolResult.iteration,
+          symbolResult.io == IoInput)
+        new SymbolStoreEntry(
+          symbolStoreKey,
+          wdlType,
+          Option(symbolResult.wdlValue).map(wdlType.coerceRawValue(_).get))
+
+      }
+    } yield symbolStoreEntries
+
+    runTransaction(action)
+  }
+
+  /** Should fail if a value is already set.  The keys in the Map are locally qualified names. */
```

Comment:
Added constraints to the database so that it will fail, and the database will stay consistent.

NOTE: Our use of MySQL forced `SYMBOL`.`ITERATION` to be non-null in the schema. MySQL refuses to validate the unique constraint on null values: https://bugs.mysql.com/bug.php?id=8173